### ISSUE TITLE
checkDiskSpace fails for Ubuntu

### ISSUE THREAD

**Issue by alexboldt** [testcontainers/testcontainers-java#45]
First off, I found this project to be extremely useful and the source code very pleasant to read :)  

This could be more of an issue with either with my configuration or with the Docker remote api than this library, but I ran into a problem with running it on Ubuntu 14.04.3, Docker version 1.9.1, build a34a1d5. 

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: 5
	at org.testcontainers.SingletonDockerClient.checkDiskSpace(SingletonDockerClient.java:172)
```

It appears that ubuntu formats its df command differently by default, if a path is too long it will insert a line break making it harder to parse. http://askubuntu.com/questions/55702/df-h-command-puts-line-breaks-in-output-how-do-i-fix 

Here is what the Docker remote api returned for me if I'm running on ubuntu:
```
Filesystem           1K-blocks      Used Available Use% Mounted on
none                   7350760   2461408   4492912  35% /
tmpfs                   379500         0    379500   0% /dev
tmpfs                   379500         0    379500   0% /sys/fs/cgroup
/dev/disk/by-uuid/76cea61b-125c-4ea2-a2b8-06ffe70c1ba6
                       7350760   2461408   4492912  35% /etc/resolv.conf
/dev/disk/by-uuid/76cea61b-125c-4ea2-a2b8-06ffe70c1ba6
                       7350760   2461408   4492912  35% /etc/hostname
/dev/disk/by-uuid/76cea61b-125c-4ea2-a2b8-06ffe70c1ba6
                       7350760   2461408   4492912  35% /etc/hosts
shm                      65536         0     65536   0% /dev/shm
tmpfs                   379500         0    379500   0% /proc/kcore
tmpfs                   379500         0    379500   0% /proc/timer_stats
```

Assuming there isn't an easy configuration fix or something on my end, when parsing that String it could ignore lines that don't have the correct number of columns. In the meantime I just locally made a fix for it.

Thanks

**Comment by rnorth** [comment]
Alex,
Thanks for trying it out, and thanks for your kind words :)

This does indeed sound like a bug and worth fixing. I imagine two changes are needed to the testcontainers code:

* Invoke `df` with the portable flag referenced in that helpful link
* Also, make the `checkDiskSpace` method more fail-safe in general. It's only there as a convenience - to warn the user about potential strange behaviour when the docker host probably lacks sufficient disk space to work properly, which seems more likely in a CI-type environment. However, aside from actual lack of space, any other unexpected failure shouldn't be allowed to cause a crash.

I'm not sure if I'll get to this right away so would happily accept a PR if you feel like it!

Thanks
Richard

**Comment by alexboldt** [comment]
Thank you! I I'll try to get a pull request out soon.

**Comment by rnorth** [comment]
Cool - I'll assign this to you then :)
Thank you.

**Comment by alexboldt** [comment]
I have a branch ready to make a pull request with, but I can't seem to push no matter what I do, keeping getting a 403. I did get an email saying I have push access now, but so far I'm not able (tried https and ssh). I can't create branches or edit files in the web UI either, not sure what is up.

**Comment by rnorth** [comment]
Hi Alex,
Thanks. We're using a fork and pull request model for this, so that's why you don't have permissions to directly push to this repo. You'll need to fork this repo in Github, though, then add your edits to a clone of _your_ repo, and finally raise a PR from there to the main project.

These pages seem to cover the process:
https://help.github.com/articles/fork-a-repo/
(Optionally) https://help.github.com/articles/creating-and-deleting-branches-within-your-repository/
https://help.github.com/articles/using-pull-requests/

I hope this is OK!
Richard

**Comment by alexboldt** [comment]
My bad thanks for the help.

**Comment by marcingrzejszczak** [comment]
I also have this issue but only in Travis - https://travis-ci.org/spring-cloud/spring-cloud-sleuth/builds/101546722#L917

I can help fixing this issue if you want to if it speeds things up ;)

**Comment by rnorth** [comment]
@marcingrzejszczak thank you :)
I've merged in Alex's PR now, though, so I hope to have a v0.99 release out soon with the fix.

**Comment by marcingrzejszczak** [comment]
great! I'm eager to check it out :)

**Comment by marcingrzejszczak** [comment]
Any chances for a Milestone release at least? :P

**Comment by rnorth** [comment]
@marcingrzejszczak sorry it's taking a while - some things in real life are keeping me busy. My goal is to release 0.99 fully today. To have greater confidence that this issue is resolved and stays resolved, and to be sure that the library continues to work on Travis, I'm setting up a Travis build for the testcontainers-java repo.

This is on top of the existing CircleCI build - perhaps a bit weird to have two CI environments for one project, but seems like a good thing to do.

Builds are running on a branch now (thanks in part to some lifted Travis config from your project ;)), and all being well I'll do the 0.99 release this afternoon. If something goes awry I'll at least try to put out an 0.99-RC1 release... Fingers crossed!

**Comment by marcingrzejszczak** [comment]
Great :) Didn't want to put any pressure on you! I've for the time being disabled those tests in Travis so don't worry.

Whatever you put there and whenever for sure will be really cool :)

**Comment by rnorth** [comment]
Released in v0.9.9 just now.
